buildSrc/src/main/kotlin/buildsrc/conventions/lang/kotlin-multiplatform.gradle.kts

```diff
@@ -126,19 +130,37 @@ kotlin.targets.withType<KotlinJvmTarget>().configureEach {
         )
     }
 
+    // Running tests.
     testRuns.configureEach {
         executionTask.configure {
             javaLauncher = javaToolchains.launcherFor {
-                languageVersion = JavaLanguageVersion(minSupportedJavaVersion)
+                languageVersion = JavaLanguageVersion(javaForTests)
             }
         }
     }
 }
 
-tasks.withType<JavaCompile>().configureEach {
+// Compiling Kotlin tests.
+tasks.named<KotlinJvmCompile>("compileTestKotlinJvm") {
+    compilerOptions {
+        jvmTarget = JvmTarget(javaForTests)
+        freeCompilerArgs.addAll(
+            "-Xjdk-release=${javaForTests.majorVersion}",
+        )
+    }
+}
+
+// Compiling Java production code.
+tasks.named<JavaCompile>("compileJava") {
     sourceCompatibility = minSupportedJavaVersion.toString()
     targetCompatibility = minSupportedJavaVersion.toString()
```

Comment:
I think we should use `options.release` property instead. It accumulates both source and target compatibilities

Reply:
I didn't know about it, thanks! Fixed in https://github.com/krzema12/snakeyaml-engine-kmp/pull/489/commits/b1edb79455a53964102e5c11c2458a3fc8bf0e31.

Reply:
There is a good talk about different flags from recent Kotlin Conf

https://youtu.be/2Vp2QeBZkfo?si=QRJNJtwbKj3uXOzl